practices/optimizations.md

```diff
@@ -0,0 +1,55 @@
+---
+slug: optimizations
+stages:
+  - development
+short_description: Optimization is the process of modifying a software system to make some aspect of it work more efficiently or use fewer resources
+tags:
+  - optimizations
+  - code coverage
+keywords:
+  - code  - quality
+  - merge
+  - testing
+  - codebase
+---
+
+# Optimizations
+
+**TL;DR**
+
+Optimization is an act, process, or methodology of making something (such as a design, system, or decision) as fully perfect, functional, or effective as possible
+
+## What Is an Optimization
+The code optimization in the synthesis phase is a program transformation technique, which tries to improve the intermediate code by making it consume fewer resources (i.e. CPU, Memory) so that faster-running machine code will result.
+
+## Why You Might Want the Optimization
+
+Optimizing an algorithm is beyond the scope of the code optimization phase. So the program is optimized. And it may involve reducing the size of the code. So optimization helps to:
+
+- Reduce the space consumed and increases the speed of compilation.
+- Manually analyzing datasets involves a lot of time. Hence we make use of software like Tableau for data analysis. Similarly manually performing the optimization is also tedious and is better done using a code optimizer.
+- An optimized code often promotes re-usability.
+
+## How to Implement the Optimization
+1. **Optimize your Code using Appropriate Algorithm**
+2. **Optimize Your Code for Memory**
+4. **Using Operators** : Most basic operations like +=, -=, and *=, when applied on basic data types could slow down the program as well. 
+5. **Problems with Functions** : For example, if you have a code like this, it might be a bad thing.
+```
+for(int i=1; i<=10; ++i)
+     DoSomething(i);
+``` 
+6. **Optimizing Loops**
+7. **Data Structure Optimization**
+
+## Common Pitfalls of the Optimization
+
```

Comment:
Please, rewrite this part in your own words. its too similar to:
_Time based optimization will give faster output, but it can lead up to a bigger code base. So to optimize the code for time performance may actually conflict with memory and size consumption. For that, you have to find the balance between time and memory, in consideration to your requirement.
Performance optimization is a never-ending process. There are always chances to improve and make your code run faster.
Sometimes, we can be tempted to use certain programming methods to run faster at the expense of not following best practices like coding standards. Try to avoid any such kind of inappropriate method_

from https://sandhill.com/article/optimizing-code-implementation-practically/

Reply:
done